src/lookupDomains.ts

```diff
@@ -0,0 +1,97 @@
+import { capture } from '@snapshot-labs/snapshot-sentry';
+import { FetchError, isSilencedError } from './addressResolvers/utils';
+import { Address, graphQlCall } from './utils';
+import { isAddress } from '@ethersproject/address';
+
+const ENS_GRAPHQL_URL = {
+  '1': 'https://api.thegraph.com/subgraphs/name/ensdomains/ens',
+  '11155111': 'https://api.studio.thegraph.com/proxy/49574/enssepolia/version/latest'
+};
+
+const DEFAULT_CHAIN_ID = '1';
+
+type Domain = {
+  name: string;
+  labelName?: string;
+  expiryDate?: number;
+};
+
+async function fetchDomainData(domain: Domain, chainId: string): Promise<Domain> {
+  const hash = domain.name.match(/\[(.*?)\]/)?.[1];
+
+  if (!hash) return domain;
+
+  const {
+    data: { data }
+  } = await graphQlCall(
+    ENS_GRAPHQL_URL[chainId],
+    `
+      query Registration {
+        registration(id: "0x${hash}") {
+          domain {
+            name
+            labelName
+          }
+        }
+      }
+    `
+  );
+  const labelName = data?.registration?.domain?.labelName;
+
+  return {
+    ...domain,
+    name: labelName ? domain.name.replace(`[${hash}]`, labelName) : domain.name
+  };
+}
+
+export default async function lookupDomains(
+  address: Address,
+  chainId = DEFAULT_CHAIN_ID
+): Promise<Address[]> {
+  if (!isAddress(address) || !ENS_GRAPHQL_URL[chainId]) return [];
```

Comment:
Maybe better to return error?

Reply:
Maybe, but all the other methods do not return any errors, and just return `[]` instead.

Reply:
i see.. ok ok 👍